### ISSUE TITLE
Call UI not opening on iPhone X

### ISSUE THREAD

**Issue by tuneerclixlogix** [voxeet/voxeet-uxkit-reactnative#25]
Device: ( iPhone X)

https://user-images.githubusercontent.com/72122766/113983731-e8387680-9867-11eb-9e8f-982d0da94797.MP4

@codlab On some iOS devices we are facing issues like this the UI opens and closes but on others it is working fine, the code on click of call button is
`VoxeetSDK.setTelecomMode(true);
const constraints = {
    audio: true,
    video: true,
    video: {
        width: {
            min: "320",
            max: "1280",
        },
        height: {
            min: "240",
            max: "720",
        },
    },
};
try {
    let createConference = await VoxeetSDK.create({ params: { dolbyVoice: true } })
    await VoxeetSDK.join(createConference.conferenceId, { constraints: constraints })
    await VoxeetSDK.invite(createConference.conferenceId, [{ externalId: id, name: name, avatarUrl: image }])
} catch (error) {
    console.log(error)
}`

**Comment by codlab** [comment]
Note that the constraints are strictly about having boolean values on mobile. It’s a probably issue here

**Comment by FabienLavocat** [comment]
Then forget about my last comment and instead use this code:
```VoxeetSDK.setTelecomMode(true);

const constraints = {
  audio: true,
  video: true,
};

try {
  let createConference = await VoxeetSDK.create({ params: { dolbyVoice: true } });

  await VoxeetSDK.join(createConference.conferenceId, { constraints: constraints });
  await VoxeetSDK.invite(createConference.conferenceId, [{ externalId: id, name: name, avatarUrl: image }]);
}
catch (error) {
  console.log(error);
}
```

**Comment by tuneerclixlogix** [comment]
This was only happening on one of our users on iPhone X , I will remove
video: {
    width: { min: "320", max: "1280", },
    height: { min: "240", max: "720", },
  },
  and then will update this, but the call UI is also not opening on receiving the call on android devices. I have another issue for that here

**Comment by tuneerclixlogix** [comment]
@FabienLavocat @codlab This issue is not resolved with the following constraints
const constraints = {
  audio: true,
  video: true,
};
I will try to manage iPhoneX to get the logs , till then can you suggest anything else ?

**Comment by tuneerclixlogix** [comment]
Seems like it is happening only on one device, we tested on other iPhone X its working fine on that.

**Comment by codlab** [comment]
@Comanga @vjard what do you think can be the issue here??

**Comment by tuneerclixlogix** [comment]
@codlab On our devices its working fine but its not working on my client's device of which i cannot retrieve the logs. Please let me know if you need any other specific log that i can share from my device .

**Comment by FabienLavocat** [comment]
Hi @tuneerclixlogix. How are you initializing the VoxeetSDK?
@vjard was able to reproduce this issue by setting the wrong application key/secret.
Please make sure your clients applications have the proper access token when initializing the SDK.

**Comment by tuneerclixlogix** [comment]
Hello @FabienLavocat , I am using the below code to initialize the sdk and the flow is like this we call an api to fetch the application key/secret from our server and on its response we run the following code
await VoxeetSDK.initialize('xxxxx', 'xxxxx')
const user = {
externalId: user_id,
name: firstName,
avatarUrl: photos[0]
}
let connectRes = await VoxeetSDK.connect(user);
let awaitingResponse = await VoxeetSDK.checkForAwaitingConference();
It's working on the other devices and we are facing this only on one device .

**Comment by FabienLavocat** [comment]
@tuneerclixlogix could you check that the SDK is initialized properly before starting the conference. I’m asking because we reproduce the exact same problem when the Consumer Key and Secret are incorrect.

**Comment by tuneerclixlogix** [comment]
@FabienLavocat Sure, is there any method i can use to check if the sdk is initialized , which i can use to verify before starting the conference ( if not initialized i will initialize it again )

**Comment by codlab** [comment]
Another possibility is that this customer is in an embargoed network where VoIP is disallowed, join request fails in such case but the UI may appear and disappear

**Comment by tuneerclixlogix** [comment]
@FabienLavocat @codlab to check this if the sdk is initialized successfully on another screen can i call the isUserLoggedIn method ? I can see that you have mentioned it as deprecated, but can i use it to verify for this scenario if the sdk is initialized properly or not ?

**Comment by codlab** [comment]
Actually, I may have a reason for the issue, can you tell where this specific user is located to?

Btw with the new implementation, you can get the events of the conferences and such which means as well that you can track why the ui is closing. Did you test that the join call succeeded as well? And was not rejected afterward?

**Comment by tuneerclixlogix** [comment]
Thanks @codlab , I think this issue has been resolved on my client's side and it seems to be not occurring now.

**Comment by codlab** [comment]
Closing this ticket since it seems to have been fixed 👍
